File: backend/smart-sales-control/src/main/java/com/sellinghouses/salescontrol/module/room/entity/Room.java
package com.sellinghouses.salescontrol.module.room.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Room {

    private Long id;

    private Long buildingId;

    private Long unitId;

    private String roomNo;

    private Integer floorNo;

    private BigDecimal area;

    private BigDecimal price;

    private String layout;

    private String orientation;

    private String decoration;

    private Integer status;

    private String remark;

    private Integer deleted;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;

    private Long createUser;

    private Long updateUser;
}

File: admin-web/src/types/room.ts
import type { PageQuery } from './common';

export interface RoomVO {
  id: number;
  buildingId: number;
  buildingName?: string;
  unitId: number;
  unitName?: string;
  roomNo: string;
  floorNo: number;
  area: number;
  price: number;
  cover?: string;
  images?: string;
  layout?: string;
  orientation?: string;
  decoration?: string;
  status: number;
  remark?: string;
  createTime?: string;
}

export interface RoomPageQuery extends PageQuery {
  buildingId?: number | '';
  unitId?: number | '';
  floorNo?: number | '';
  roomNo?: string;
  status?: number | '';
}

export interface RoomForm {
  id?: number;
  buildingId?: number;
  unitId?: number;
  roomNo: string;
  floorNo?: number;
  area?: number;
  price?: number;
  cover?: string;
  images?: string;
  layout?: string;
  orientation?: string;
  decoration?: string;
  status: number;
  remark?: string;
}

export interface RoomPriceUpdate {
  id: number;
  price: number;
}

export interface RoomStatusUpdate {
  id: number;
  status: number;
}
